src/OpenTracing/TracerSpanBuilderExtensions.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace OpenTracing
+{
+    using System.Linq;
+
+    public static class TracerSpanBuilderExtensions
+    {
+        public static ISpanBuilder BuildSpan(ITracer tracer, string operationName)
+        {
+            return new SpanBuilder(tracer, operationName);
+        }
+
+        private sealed class SpanBuilder : ISpanBuilder
+        {
+            private readonly ITracer tracer;
+
+            private string operationName;
+            private ICollection<Tuple<string, ISpanContext>> references;
+            private DateTimeOffset? explicitStartTime;
+            private IDictionary<string, object> tags;
+
+            public SpanBuilder(ITracer tracer, string operationName)
```

Comment:
Some implementations may forgo initialization logic if tracing on the parent span is disabled (to enable hot path code).

Three obvious solutions to this:
1. Users can implement their own alternative extension methods
2. Users can implement ALL possible extension methods (just remove ISpanBuilder entirely from this interfaces project
3. Leave as is, requiring users to implement ISpanBuilder